web/src/ehr/transfer/NewTransfer.vue

```diff
@@ -34,10 +44,56 @@ export default {
     }
   },
   methods: {
+    checkForm() {
+      // Reset the formErrors array:
+      this.formErrors.length = 0
+
+      if (this.transfer.transferDate === null) {
+        this.formErrors.push("The transfer date is a required field.")
+      }
+
+      if (this.transfer.carePlan.patientProblems.length === 0) {
+        this.formErrors.push("The care plan should have at least 1 problem.")
+      }
+
+      this.transfer.carePlan.patientProblems.forEach((patientProblem, i) => {
+        var hasInterventions = false
+        patientProblem.interventions.forEach((intervention, j) => {
+          // dont perform the check for the empty intervention placeholder
+          if ((i > 0 || j != 0) && j == patientProblem.interventions.length - 1) {
+            return
+          }
+          if (intervention.comment == "") {
+            this.formErrors.push("Each intervention must have a comment.")
+          } else {
+            hasInterventions = true
+          }
+        })
+        // Skip the check when:
+        if ( // Its the placeholder, so last problem, but not the only one
+            i != 0 && i == this.transfer.carePlan.patientProblems.length - 1 &&
+            // and it does not have any interventions
+            !hasInterventions
+        ) {
+          return
+        }
+        if (patientProblem.problem.name === "") {
+          this.formErrors.push("Each problem must have a description.")
+          return false
+        }
+        if (patientProblem.interventions.length == 0) {
+          this.formErrors.push("Each problem must have at least 1 intervention.")
+        }
+      })
+      return this.formErrors.length == 0
+    },
```

Comment:
Can this be de-spaghettified? It's really hard to read. Maybe something like this (untested):

```

      this.transfer.carePlan.patientProblems.forEach((patientProblem, i) => {
        // skip problems placeholder
        if (this.transfer.carePlan.patientProblems.length - 1 === i &&
            !patientProblem.problem.name &&
            patientProblem.interventions.filter(intervention => !!intervention.comment).length === 0) {
          return;
        }

        // skip the intervention placeholder
        patientProblem.interventions.slice(0, patientProblem.interventions.length - 1).forEach(intervention => {
          if (!intervention.comment) {
            this.formErrors.push("Each intervention must have a comment.")
          }
        });

        if (!patientProblem.problem.name) {
          this.formErrors.push("Each problem must have a description.")
        }
        
        if (patientProblem.interventions.length === 0) {
          this.formErrors.push("Each problem must have at least 1 intervention.")
        }
      });
```

Reply:
This gives me headaches. Is it really cleaner?

Reply:
Going back to the original source the issue I have is that it takes a while to understand what the code does. At also needs 4 comments to make sense. At the same time it's a simple algorithm but it contains a lot of exceptions (error when empty but not if it's the last or first section) which makes it hard to keep the code simple.

What I tried to do is to get rid of things like this:

```js
if ((i > 0 || j != 0) && j == patientProblem.interventions.length - 1) {
```

An alternative could be to get rid of the placeholders before validation (basically copy the data, delete the placeholders then validate). That would make it a lot more straightforward and we can drop checking for empty interventions in the backend. I don't want to make this a big deal as it's a simple validation used by a single component but I would like the code to be a little bit cleaner.

What do you think?